File: typescript/fastify/src/app/modules/clients/dtos/inputs/findByCPF-client.dto.ts
import { IsString, Validate } from "class-validator";
import { Transform } from "class-transformer";

// Utils
import { cleanDigits } from "@/app/common/helpers/cleanDigits.helper";

// Validator
import { CPFValidator } from "@/app/common/validators/cpf.validator";

// Messages
import { MESSAGES } from "@/app/common/messages";

export class FindByCPFClientDto {
  @IsString({
    message: MESSAGES.validation.IsString("cpf"),
  })
  @Validate(CPFValidator, {
    message: MESSAGES.validation.InvalidCPF("cpf"),
  })
  @Transform((params) => cleanDigits(params.value))
  cpf: string;
}

File: rust/actix/src/modules/client/dto/find_by_cpf_client_dto.rs
use serde::{Deserialize, Serialize};
use validator::Validate;
use crate::common::validator::validate_cpf;
use crate::common::helper::clean_digits::clean_digits;

#[derive(Debug, Deserialize, Serialize, Validate)]
pub struct FindByCPFClientDto {
    #[validate(custom(function = "validate_cpf"))]
    pub cpf: String,
}

impl FindByCPFClientDto {
    pub fn transform(&mut self) {
        // CPF
        self.cpf = clean_digits(&self.cpf);
    }
}
